Composer/packages/client/src/CreationFlow/DefineConversation/index.js

```diff
@@ -59,28 +92,22 @@ export function DefineConversation(props) {
     });
   };
 
-  //disable the next button if the text has errors.
-  const getErrorMessage = text => {
-    if (typeof onGetErrorMessage === 'function') {
-      const result = onGetErrorMessage(text);
-      if (result === '' && disable) {
-        setDisable(false);
-      }
-
-      if (result !== '' && !disable) {
-        setDisable(true);
-      }
-
-      return result;
-    } else {
-      return '';
-    }
-  };
-
-  // // update the path in the form and toggle the location picker.
-  const updateLocation = path => {
-    updateForm('location')(null, path);
-  };
+  function getDefaultName() {
+    let i = -1;
+    const bot = templateId || botName;
+    let defaultName = '';
+
+    do {
+      i++;
+      defaultName = `${bot}-${i}`;
+    } while (
+      ~bots.findIndex(bot => {
+        return bot.name === defaultName;
+      }) &&
+      i < MAXTRYTIMES
+    );
+    return defaultName;
```

Comment:
So if `MAXTRYTIMES` is reached without finding a unique name, the default name will be the last computed name and invalid right?

Reply:
Previously, it will be an endless loop if we cannot find an unique name(probably due to a bug). According to @compulim, we should throw an error when the MAXTIMES reached. But I feel even if this do happen, we should not decrease user's experience. This is a small feature. So I keep the code this way just like what you said. If the code generate a duplicated name, an error message will show.